Sources/Vehicle/Templates/CarPlayActionsTemplate.swift

```diff
@@ -0,0 +1,107 @@
+import CarPlay
+import Foundation
+import RealmSwift
+import Shared
+
+@available(iOS 15.0, *)
+final class CarPlayActionsTemplate: CarPlayTemplateProvider {
+    private var actionsToken: NotificationToken?
+    private let actions: Results<Action>
+
+    private var noActionsView: CPInformationTemplate = {
+        CPInformationTemplate(
+            title: L10n.About.Logo.title,
+            layout: .leading,
+            items: [
+                .init(title: L10n.CarPlay.NoActions.title, detail: nil),
+            ],
+            actions: []
+        )
+    }()
+
+    var template: CPTemplate
+
+    weak var interfaceController: CPInterfaceController?
+
+    init(actions: Results<Action>) {
+        self.actions = actions
+        self.template = CPTemplate()
+    }
+
+    func templateWillDisappear(template: CPTemplate) {
+        if template == self.template {
+            actionsToken?.invalidate()
+        }
+    }
+
+    func templateWillAppear(template: CPTemplate) {
+        if template == self.template {
+            update()
+        }
+    }
+
+    func update() {
+        updateList(for: actions)
+        template.tabTitle = L10n.Carplay.Navigation.Tab.actions
+        template.tabImage = MaterialDesignIcons.lightningBoltIcon.image(
+            ofSize: .init(width: 64, height: 64),
+            color: nil
+        )
+        template.tabSystemItem = .more
+    }
+
+    func updateList(for actions: Results<Action>) {
+        actionsToken?.invalidate()
+        actionsToken = actions.observe { [weak self] _ in
+            self?.updateActions(actions: actions)
+        }
+
+        if let listTemplate = template as? CPListTemplate {
+            listTemplate.updateSections([section(actions: actions)])
+        } else {
+            template = CPListTemplate(
```

Comment:
since the template was taken in the updateTemplates method in the scene delegate, how does modifying it here trigger a re-set in the tab bar?

Reply:
Updated it, it shouldn't matter anymore